src/pages/Search/SearchAdvancedFiltersPage/SearchFiltersExportedPage/index.tsx

```diff
@@ -0,0 +1,104 @@
+
+    // Get all of the initial values from the advanced filters form and store them locally so we can modify
+    // them without mutating the original form
+    const initialState = useMemo(() => {
+        return Object.values(CONST.SEARCH.DATE_MODIFIERS).reduce((acc, dateType) => {
+            acc[dateType] = searchAdvancedFiltersForm?.[`${CONST.SEARCH.SYNTAX_FILTER_KEYS.EXPORTED}${dateType}`] ?? null;
+            return acc;
+        }, {} as SearchFiltersExportedPageValues);
+        // The initial value of a useState only gets calculated once, so we dont need to keep calculating this initial state
+        // eslint-disable-next-line react-compiler/react-compiler
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    const [view, setView] = useState<SearchDateModifier | null>(null);
+    const [localDateValues, setLocalDateValues] = useState<SearchFiltersExportedPageValues>(initialState);
+
+    const setDateValue = (key: SearchDateModifier, dateValue: string | null) => {
+        setLocalDateValues((currentValue) => {
+            // If we are setting the 'on' to 'never', reset the other dates
+            if (key === CONST.SEARCH.DATE_MODIFIERS.ON && dateValue === CONST.SEARCH.NEVER) {
+                return {
+                    [CONST.SEARCH.DATE_MODIFIERS.ON]: dateValue,
+                    [CONST.SEARCH.DATE_MODIFIERS.AFTER]: null,
+                    [CONST.SEARCH.DATE_MODIFIERS.BEFORE]: null,
+                };
+            }
+
+            // If we are setting any other value while 'on' is set to 'never', reset 'on' to null
+            if (key !== CONST.SEARCH.DATE_MODIFIERS.ON && currentValue?.[CONST.SEARCH.DATE_MODIFIERS.ON] === CONST.SEARCH.NEVER) {
+                return {
+                    ...currentValue,
+                    [key]: dateValue,
+                    [CONST.SEARCH.DATE_MODIFIERS.ON]: null,
+                };
+            }
+
+            return {
+                ...currentValue,
+                [key]: dateValue,
+            };
+        });
+    };
+
+    const navigateToRootView = useCallback(() => {
+        setView(null);
+    }, []);
+
+    const resetChanges = useCallback(() => {
+        // Reset each field back to null
+        setLocalDateValues((prev) => {
+            return Object.fromEntries(Object.entries(prev).map(([key]) => [key, null])) as SearchFiltersExportedPageValues;
+        });
+    }, []);
+
+    const applyChanges = useCallback(() => {
+        updateAdvancedFilters({
+            [`${CONST.SEARCH.SYNTAX_FILTER_KEYS.EXPORTED}${CONST.SEARCH.DATE_MODIFIERS.ON}`]: localDateValues?.[CONST.SEARCH.DATE_MODIFIERS.ON],
+            [`${CONST.SEARCH.SYNTAX_FILTER_KEYS.EXPORTED}${CONST.SEARCH.DATE_MODIFIERS.AFTER}`]: localDateValues?.[CONST.SEARCH.DATE_MODIFIERS.AFTER],
+            [`${CONST.SEARCH.SYNTAX_FILTER_KEYS.EXPORTED}${CONST.SEARCH.DATE_MODIFIERS.BEFORE}`]: localDateValues?.[CONST.SEARCH.DATE_MODIFIERS.BEFORE],
```

Comment:
```suggestion
            exportedOn: localDateValues.On,
            exportedAfter: localDateValues.After,
            exportedBefore: localDateValues.Before,
```

I think the overuse of our CONST make things more complex than they should be. We already have well defined types here, so we can do it as above and it is safe. Any keys that is poorly types would be catch by typescript.

<img width="474" alt="image" src="https://github.com/user-attachments/assets/810b9e70-2992-4eb9-b55d-2c10ca738081" />